module/VuFind/src/VuFind/Db/Service/AccessTokenService.php

```diff
@@ -92,7 +93,8 @@ public function getByIdAndType(
                            ->setType($type)
                            ->setCreated(new DateTime());
         }
-
+        
+        $this->persistEntity($result);
```

Comment:
I think you should move this inside the `if ($result === null && $create) {` clause. You should only persist the data if you've been asked to create it. Otherwise you may perform unnecessary writes and/or illegally pass null to the persist method.